### PR TITLE
GraphQL Binding Doc

### PR DESCRIPTION
**Pull request by amirmm11**
**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Documentation for the GraphQL binding

## Issue reference
https://github.com//issues/1612

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+kind: Component
+metadata:
+  name: example.bindings.graphql
+spec:
+  type: bindings.graphql
+  version: v1
+  metadata:
+    - name: endpoint
+      value:  #required    
+    - name: header:xyz # e.g. header:x-hasura-access-key
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md]: Should the name be `header` and the value be `xyz`? Why is value currently blank>

File: daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: example.bindings.graphql
+spec:
+  type: bindings.graphql
+  version: v1
+  metadata:
+    - name: endpoint
+      value:  #required
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md]:
```suggestion
      value: "http://localhost:4000/graphql" #required    
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+{{% alert title="Warning" color="warning" %}}
+The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Binding support |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| endpoint | Y | Output | GraphQL endpoint string See [here](#url-format) for more details | `"http://localhost:4000/graphql"` |
+| header | N | Output | GraphQL header | `"header:Cache-Control"` |
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md]: Here it looks like value is `"header:Cache-Control"`, whereas above you have it as the name. Please update with correct syntax.

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md (hunk 1 above)
Comment (amirmm11) [reply daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md]: No, please see https://github.com/dapr/components-contrib/pull/844#issuecomment-861009315

**Comment by amirmm11** [comment]
@AaronCrawfis Updated the document per your recommendations.

File: daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md (hunk 1 above)
Comment (AaronCrawfis) [reply daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md]: Oh perfect, thanks for the clarification!

File: daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{{% alert title="Warning" color="warning" %}}
+The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Binding support |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| endpoint | Y | Output | GraphQL endpoint string See [here](#url-format) for more details | `"http://localhost:4000/graphql/graphql"` |
+| header | N | Output | GraphQL header | `"no-cache"` |
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/components-reference/supported-bindings/graghql.md]:
```suggestion
| header:[HEADERKEY] | N | Output | GraphQL header. Specify the header key in the `name`, and the header value in the `value`. | `"no-cache"` (see above) |
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
One last small clarification and we'll be good to go. Thanks!

**Comment by amirmm11** [comment]
Thanks, @AaronCrawfis, for your suggestions. I pushed your changes.

**Review by AaronCrawfis — APPROVED** [review]

**Comment by AaronCrawfis** [comment]
@amirmm11 what is the PR in components-contrib that pertains to the new binding?

**Comment by yaron2** [comment]
> @amirmm11 what is the PR in components-contrib that pertains to the new binding?

https://github.com/dapr/components-contrib/pull/844

**Pull request merged**